### DISCUSSION TITLE
Redirecting on Azure Storage blob has XHR error

### DISCUSSION THREAD

**Discussion by imtrobin** [YOURLS/YOURLS#3011] [Q&A]
Hi,

I'm using yourls to redirect to a file on Azure Blob Storage (CORS allowed for all). It gets an error, and I tested with test-cors.org, it gives an XHR event: error

When I use tinyurl, I can redirect fine.

More information here
https://docs.microsoft.com/en-us/answers/questions/494882/azure-storage-get-not-working-with-url-shortner-co.html

**Comment by dgw** [comment]
Needs more info on what "an error" is, probably found in the browser console. There's [a whole MDN page about CORS errors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors), but I can't even begin to speculate without seeing an error message. JS can't access _why_ a CORS error occurred for security reasons, so that test site that looks for XHR events (which you screenshotted for the Microsoft answers site) can't show you more than "error".

**Comment by imtrobin** [comment]
I'm still learning about how CORS works. Using a normal webbrowser, it works fine. I saw a yellow warning CORS block but disappears quickly

 When using http://test-cors.org, which only send the GET once, it has in the log
 
 Access to XMLHttpRequest at 'http://xxxx.com/xxxx' from origin 'http://test-cors.org' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
 
 Tinyurl sends response with 307 Internal Redirect with Referrer Policy: strict-origin-when-cross-origin and then a 301 with access-control headers (so I think the browser retry again) , where yourls just response with 302
